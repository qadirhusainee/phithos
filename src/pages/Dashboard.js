import React, { useState } from "react";
import type { FunctionComponent } from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Button,
  Image,
  SafeAreaView,
} from "react-native";
import { useSelector } from "react-redux";

import Baner from "../component/Baner";
import authReducer from "../redux/reducers/authReducer";

const Dashboard: FunctionComponent = () => {
  const { email } = useSelector((state) => state.auth.userDetails);
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <Baner />
      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
        <Text>{email} </Text>
      </View>
    </SafeAreaView>
  );
};

export default Dashboard;
