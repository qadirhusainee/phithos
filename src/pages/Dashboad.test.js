import React from "react";
import renderer from "react-test-renderer";
import { render } from "react-native-testing-library";

import Dashboard from "./Dashboard";

jest.mock("react-redux", () => {
  const ActualReactRedux = require.requireActual("react-redux");
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return { email: "qadirhusainee@gmail.com" };
    }),
  };
});

describe("Dashboard", () => {
  it("Matches Snapshot", () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Showing Baner", () => {
    const { queryByText } = render(<Dashboard />);

    const element = queryByText("Emulator");
    expect(element).toBeDefined();
  });

  it("Showing Email", () => {
    const { queryByText } = render(<Dashboard />);

    const element2 = queryByText("qadirhusainee@gmail.com");
    expect(element2).toBeDefined();
  });
});
