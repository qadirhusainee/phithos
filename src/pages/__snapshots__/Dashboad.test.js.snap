// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Matches Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "grey",
        "borderWidth": 0.5,
        "height": 80,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingTop": 5,
        }
      }
    >
      Simulator(IOS)
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginRight": 20,
          "width": 750,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      qadirhusainee@gmail.com
       
    </Text>
  </View>
</RCTSafeAreaView>
`;
